instrumentation/que/lib/opentelemetry/instrumentation/que/tag_getter.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Que
+      # The TagGetter class provides methods for reading tracing information
+      # from Que tags.
+      #
+      # @example
+      #   OpenTelemetry.propagation.extract(tags, getter: TagGetter.new)
+      class TagGetter
+        def get(carrier, key)
+          tags_array_to_hash(carrier)[key]
+        end
+
+        def keys(carrier)
+          tags_array_to_hash(carrier).keys
+        end
+
+        private
+
+        def tags_array_to_hash(tags)
```

Comment:
This is rough 😞 - the API really isn't designed well for this sort of data format. I think you'd be better off with the original code (converting to a hash and using the default getter) for the `extract` operation. Doing it this way is just adding abstraction for no value - we're still allocating the hash, we're just hiding it.

Reply:
yes, I agree. It felt somewhat odd. Reverted it: https://github.com/open-telemetry/opentelemetry-ruby/compare/b828d9146b67d623f84b6b963c7b8bd8abb302ee..1723d99922224afec6a1f1b9370f02feada1a0e3